go/ai/generate.go

```diff
@@ -62,18 +54,24 @@ func DefineModel(
 		// Always make sure there's at least minimal metadata.
 		metadata = &ModelMetadata{
 			Label: name,
+			Info: ModelInfo{
+				Label:    name,
+				Supports: &ModelInfoSupports{},
+			},
+			Versions: []string{},
 		}
 	}
 	if metadata.Label != "" {
 		metadataMap["label"] = metadata.Label
 	}
 	supports := map[string]bool{
-		"media":      metadata.Supports.Media,
-		"multiturn":  metadata.Supports.Multiturn,
-		"systemRole": metadata.Supports.SystemRole,
-		"tools":      metadata.Supports.Tools,
+		"media":      metadata.Info.Supports.Media,
```

Comment:
Unmarshal `ModelInfoSupports` into the map programatically so we have full coverage of fields going forward.

Reply:
For the record: After internal conversation with Alex, it was decided not to follow this suggestion since these changes will be breaking current functionality: `DefineModel` would have to check for errors during Marshal/Unmarshal process and return value must change to `(Model, error)` which will break the current state of the API.